airflow/io/path.py

```diff
@@ -94,6 +136,15 @@ def samestore(self, other: typing.Any) -> bool:
             and self.storage_options.get("conn_id") == other.storage_options.get("conn_id")
         )
 
+    @property
+    def store(self) -> ObjectStore:
+        return ObjectStore(
+            protocol=self.protocol,
+            conn_id=self.storage_options.get("conn_id"),
+            fs=self.fs,
+            storage_options=self._storage_options,
+        )
```

Comment:
Not sure if this is correct cc @bolkedebruin

Reply:
Mmm yes. This looks a bit fishy at first sight: 

* requesting the property instantiates the ObjectStore repeatably 
* Store is kind of ephemeral 
* What is the need for it as the information is already in the url?
* storage_options can contain sensitive information (passwords etc)

The the basic question why is it needed and can it be done differently?

Reply:
I using it to provide context for given AIP-60 URI - that would allow us to possibly get additional info about that path - but that in general would be used in case of hooks, as `ObjectStore` would be a rough equivalent of particular hook. 

I've removed that part, and will just pass `ObjectStoragePath` as that context.